src/corner_kick/src/containers/Canvas/__helpers__/shapes.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * This file contains helper functions for unit testing
+ */
+
+import * as Chance from 'chance';
+
+import { BYTES_PER_SHAPE } from 'SRC/constants/canvas';
+import { IShape } from 'SRC/types';
+
+const chance = new Chance();
+
+/**
+ * Generates a random number of shapes for testing purposes
+ * @param count Number of shapes to generate
+ */
+export const generateRandomShapes = (count: number): IShape[] => {
+    return new Array(count).fill(true).map(() => {
+        return {
+            texture: chance.integer({ min: 0, max: 255 }),
+            flags: chance.integer({ min: 0, max: 255 }),
+            x: chance.integer({ min: -32768, max: 32767 }),
+            y: chance.integer({ min: -32768, max: 32767 }),
+            width: chance.integer({ min: -32768, max: 32767 }),
+            height: chance.integer({ min: -32768, max: 32767 }),
+            rotation: chance.integer({ min: -32768, max: 32767 }),
+            opacity: chance.integer({ min: 0, max: 255 }),
+            red: chance.integer({ min: 0, max: 255 }),
+            green: chance.integer({ min: 0, max: 255 }),
+            blue: chance.integer({ min: 0, max: 255 }),
+        };
+    });
+};
+
+/**
+ * Converts a layer into a binary ArrayBuffer. Used for testing purposes.
+ * @param layer layer number
+ * @param flags layer flags
+ * @param shapes layer shapes
+ */
+export const generateLayerBuffer = (layer: number, flags: number, shapes: IShape[]) => {
+    const arrayBuffer = new ArrayBuffer(2 + shapes.length * BYTES_PER_SHAPE);
+    const dataView = new DataView(arrayBuffer);
+
+    dataView.setUint8(0, layer);
+    dataView.setUint8(1, flags);
+
+    shapes.forEach((shape, index) => {
+        const startPos = 2 + index * BYTES_PER_SHAPE;
+        dataView.setUint8(startPos, shape.texture);
```

Comment:
I believe on CPP side we're sending a uniform 32 bits per attribute.  We do this by making an vector of int32s, and then add the vector to the buffer to be sent. This is so that all values can fit into the vector without special bit shifting and packing.

@garethellis0 thoughts?

Reply:
Ah, so we have a few options here. 
- Send over a `std::vector<Shape>`? If the functions we use will accept this and the struct is ordered properly with the correct types it should work? Downside is it's not very explicit, which opens the door to hard-to-track bugs
- Convert each `struct` instance into an `std::bitset<N>`, and then send over a `std::vector<std::bitset<N>>`, where `N` is known at compile-time (probably do like `typedef ShapeBitset std::bitset<NUM_BITS_IN_SHAPE>`). 
- The final kindy hacky option is to send over an `std::vector<bool>`

I favor the 2nd option here.

Reply:
Alternatively, we can change the visualizer to accept 32 bit for each number. Downside is the size to describe each shape increases from 16 bytes to 32 bytes.

Reply:
Assuming that you're receiving as a bit stream, then we could also use 16bit integer arrays and just pack the 8-bit attributes to one, and split the 32-bit attributes into two. As long as the order is correct, it's fine right?

i.e.
8+8 for the layer overhead (layer number, flags)
8+8 for the shape texture and flags
16 for x, y, w, h, rotation
32/2 for alpha and red
32/2 for green and blue

Reply:
Yup, as long as the order remains the same, it should be fine

Reply:
Ok, it's done. Please take a look in #445